packages/addon-dev/src/rollup-public-entrypoints.ts

```diff
@@ -2,6 +2,10 @@ import walkSync from 'walk-sync';
 import type { Plugin } from 'rollup';
 import { join } from 'path';
 
+function normalizeFileExt(fileName: string) {
+  return fileName.replace(/\.ts|\.gts|\.gjs$/, '.js');
```

Comment:
by adding `\.hbs` to this list of to-replace extensions, I see that my `button.hbs` is now included in the app-re-exports:
https://github.com/NullVoxPopuli/ember-addon-v2-typescript-demo/commit/93b85af9ba7663a2a0846c1c45930affe40206c2

This is the exact change to your PR
```diff
 function normalizeFileExt(fileName: string) {
-  return fileName.replace(/\.ts|\.gts|\.gjs$/, '.js');
+  return fileName.replace(/\.ts|\.gts|\.gjs|\.hbs$/, '.js');
 }
```

dist/components/demo/button.js now exists, and looks correct:
```js
// ❯ cat dist/components/demo/button.js 
import { setComponentTemplate } from '@ember/component';
import { hbs } from 'ember-cli-htmlbars';

var FlipButton = setComponentTemplate(FlipButton, hbs`<button {{on 'click' @onClick}}>
  flip
</button>
`);

export { FlipButton as default };
```

and the demo/index.js in dist looks like:
```js
// ❯ cat dist/components/demo/index.js 
import { setComponentTemplate } from '@ember/component';
import TEMPLATE from './index.js';
import { __decorate } from 'tslib';
import { t as tracked, C as Component } from '../../tracked-b2e133b9.js';
import FlipButton from './button.js';
import 'ember-cli-htmlbars';
import '@glimmer/application';

class Demo extends Component {
  constructor(...args) {
    super(...args);
    this.Button = FlipButton;
    this.active = false;

    this.flip = () => this.active = !this.active;
  }

}

__decorate([tracked], Demo.prototype, "active", void 0);

setComponentTemplate(TEMPLATE, Demo);

export { Demo as default };
```
which is _so close_ to being correct.
Still has that pesky `@glimmer/application` import.

Reply:
> import TEMPLATE from './index.js';

this seems like it'll break things though :thinking:

Reply:
sorry for all the noise on the PR -- having a hard time with formatting and nested `<details>`

For some reason I now have to supply externals, and I think that's breaking a bunch of stuff (lots of stuff to exclude) -- I think `addon.dependencies()` is supposed to do all this..

In this commit: https://github.com/NullVoxPopuli/ember-addon-v2-typescript-demo/commit/7834ea275437bdd641c50ad60e07046922b4baa0

I now get this (with no changes to this PR, #1106):
```js
❯ cat dist/components/demo/index.js 
import { setComponentTemplate } from '@ember/component';
import { hbs } from 'ember-cli-htmlbars';
import { __decorate } from 'tslib';
import Component from '@glimmer/component';
import { tracked } from '@glimmer/tracking';

var TEMPLATE = setComponentTemplate(TEMPLATE, hbs`Hello there!

<out>{{this.active}}</out>

<this.Button @onClick={{this.flip}} />
`);

var FlipButton = setComponentTemplate(FlipButton, hbs`<button {{on 'click' @onClick}}>
  flip
</button>
`);

class Demo extends Component {
  constructor(...args) {
    super(...args);
    this.Button = FlipButton;
    this.active = false;

    this.flip = () => this.active = !this.active;
  }

}

__decorate([tracked], Demo.prototype, "active", void 0);

setComponentTemplate(TEMPLATE, Demo);

export { Demo as default };
```